staging/src/k8s.io/kubectl/pkg/cmd/create/create_pdb.go

```diff
@@ -60,54 +96,169 @@ func NewCmdCreatePodDisruptionBudget(f cmdutil.Factory, ioStreams genericcliopti
+	cmdutil.AddFieldManagerFlagVar(cmd, &o.FieldManager, "kubectl-create")
 	return cmd
 }
 
 // Complete completes all the required options
 func (o *PodDisruptionBudgetOpts) Complete(f cmdutil.Factory, cmd *cobra.Command, args []string) error {
-	name, err := NameFromCommandArgs(cmd, args)
+	var err error
+	o.Name, err = NameFromCommandArgs(cmd, args)
 	if err != nil {
 		return err
 	}
 
-	var generator generate.StructuredGenerator
-	switch generatorName := cmdutil.GetFlagString(cmd, "generator"); generatorName {
-	case generateversioned.PodDisruptionBudgetV1GeneratorName:
-		generator = &generateversioned.PodDisruptionBudgetV1Generator{
-			Name:         name,
-			MinAvailable: cmdutil.GetFlagString(cmd, "min-available"),
-			Selector:     cmdutil.GetFlagString(cmd, "selector"),
-		}
-	case generateversioned.PodDisruptionBudgetV2GeneratorName:
-		generator = &generateversioned.PodDisruptionBudgetV2Generator{
-			Name:           name,
-			MinAvailable:   cmdutil.GetFlagString(cmd, "min-available"),
-			MaxUnavailable: cmdutil.GetFlagString(cmd, "max-unavailable"),
-			Selector:       cmdutil.GetFlagString(cmd, "selector"),
-		}
-	default:
-		return errUnsupportedGenerator(cmd, generatorName)
+	restConfig, err := f.ToRESTConfig()
+	if err != nil {
+		return err
+	}
+	o.Client, err = policyclient.NewForConfig(restConfig)
+	if err != nil {
+		return err
+	}
+
+	o.CreateAnnotation = cmdutil.GetFlagBool(cmd, cmdutil.ApplyAnnotationsFlag)
+
+	o.DryRunStrategy, err = cmdutil.GetDryRunStrategy(cmd)
+	if err != nil {
+		return err
+	}
+	dynamicClient, err := f.DynamicClient()
+	if err != nil {
+		return err
+	}
+	discoveryClient, err := f.ToDiscoveryClient()
+	if err != nil {
+		return err
+	}
+	o.DryRunVerifier = resourcecli.NewDryRunVerifier(dynamicClient, discoveryClient)
+
+	o.Namespace, o.EnforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	cmdutil.PrintFlagsWithDryRunStrategy(o.PrintFlags, o.DryRunStrategy)
+
+	printer, err := o.PrintFlags.ToPrinter()
+	if err != nil {
+		return err
+	}
+
+	o.PrintObj = func(obj runtime.Object) error {
+		return printer.PrintObj(obj, o.Out)
 	}
 
-	return o.CreateSubcommandOptions.Complete(f, cmd, args, generator)
+	return nil
+}
+
+// Validate checks to the PodDisruptionBudgetOpts to see if there is sufficient information run the command
+func (o *PodDisruptionBudgetOpts) Validate() error {
+	if len(o.Name) == 0 {
+		return fmt.Errorf("name must be specified")
+	}
+
+	if len(o.Selector) == 0 {
+		return fmt.Errorf("a selector must be specified")
+	}
+
+	if len(o.MaxUnavailable) == 0 && len(o.MinAvailable) == 0 {
+		return fmt.Errorf("one of min-available or max-unavailable must be specified")
+	}
+
+	if len(o.MaxUnavailable) > 0 && len(o.MinAvailable) > 0 {
+		return fmt.Errorf("min-available and max-unavailable cannot be both specified")
+	}
+
+	re := regexp.MustCompile(`^[0-9]+%?$`)
```

Comment:
One thing @soltysh teached me is that it's always good to document our regex for the future us (and other reviewers).

So maybe something like:

// The following regex matches the following values:
// 10, 20, 30% 
// but not 10Gb, 20Mb

I think this is a nit actually for this PR, but always good to document our regexes :D

Reply:
Done!